cmd/zfs_object_agent/src/server.rs

```diff
@@ -39,6 +38,47 @@ impl Server {
         Ok(nvl)
     }
 
+    pub fn ustart(connection: UnixStream) {
+        let (r, w) = connection.into_split();
+        let mut server = Server {
+            input: r,
+            output: Arc::new(Mutex::new(w)),
+            pool: None,
+            num_outstanding_writes: Arc::new(std::sync::Mutex::new(0)),
+        };
+        tokio::spawn(async move {
+            loop {
+                let nvl = match tokio::time::timeout(
+                    Duration::from_millis(1000),
+                    Self::get_next_request(&mut server.input),
+                )
```

Comment:
```suggestion
                let nvl = Self::get_next_request(&mut server.input)
```
The timeout in `start()` is a giant hack because the kernel doesn't send flush-writes commands when it needs to.  No need to replicate it here.